src/ui/WindowViewModelBase.hh

```diff
@@ -17,7 +19,65 @@ struct Size
     int Height{};
 };
 
+enum class DialogResult
+{
+    None = 0,
+    OK,
+    Cancel,
+    Yes,
+    No,
+    Retry,
+};
+
+class WindowViewModelBase : public ViewModelBase
+{
+public:
+    /// <summary>
+    /// The <see cref="ModelProperty" /> for the window title.
+    /// </summary>
+    static StringModelProperty WindowTitleProperty;
+
+    /// <summary>
+    /// Gets the window title.
+    /// </summary>
+    const std::wstring& GetWindowTitle() const { return GetValue(WindowTitleProperty); }
+
+    /// <summary>
+    /// Sets the window title.
+    /// </summary>
+    void SetWindowTitle(const std::wstring& sTitle) { SetValue(WindowTitleProperty, sTitle); }
```

Comment:
Three lines per property - the first declares the ModelProperty (implementation in cpp file). The second and third provide ways to read/write the value - these call into `ViewModelBase` for managing the buckets. To make a property read-only, the setter method can be made protected or private.